src/py/flwr/client/process/process.py

```diff
@@ -0,0 +1,118 @@
+from flwr.proto.appio_pb2_grpc import ClientAppIoStub, add_ClientAppIoServicer_to_server
+
+# pylint: disable=E0611
+from flwr.server.superlink.fleet.grpc_bidi.grpc_server import generic_create_grpc_server
+
+from .clientappio_servicer import ClientAppIoServicer
+from .utils import _get_load_client_app_fn
+
+
+def _run_background_client(  # pylint: disable=R0914
+    address: str,
+    token: int,
+) -> None:
+    """Run background Flower ClientApp process."""
+
+    def on_channel_state_change(channel_connectivity: str) -> None:
+        """Log channel connectivity."""
+        log(DEBUG, channel_connectivity)
+
+    channel = create_channel(
+        server_address=address,
+        insecure=True,
+    )
+    channel.subscribe(on_channel_state_change)
+
+    try:
+        stub = ClientAppIoStub(channel)
+
+        req = PullClientAppInputsRequest(token=token)
+        res = stub.PullClientAppInputs(req)
+        # fab_file = res.fab
+        run = run_from_proto(res.run)
+        message = message_from_proto(res.message)
+        context = context_from_proto(res.context)
+        # Ensures FAB is installed (default is Flower directory)
+        # install_from_fab(
+        #     fab_file, None, True
+        # )
+        load_client_app_fn = _get_load_client_app_fn(
+            default_app_ref="",
+            project_dir="",
+            multi_app=True,
+            flwr_dir=None,
+        )
+        # print(f"FAB ID: {run.fab_id}, FAB version: {run.fab_version}")
+        client_app: ClientApp = load_client_app_fn(
+            run.fab_id, run.fab_version  # To be optimized later
+        )
+        # Execute ClientApp
+        reply_message = client_app(message=message, context=context)
+
+        proto_message = message_to_proto(reply_message)
+        proto_context = context_to_proto(context)
+        req = PushClientAppOutputsRequest(
+            token=token,
+            message=proto_message,
+            context=proto_context,
+        )
+        res = stub.PushClientAppOutputs(req)
+    except KeyboardInterrupt:
+        log(INFO, "Closing connection")
+    except grpc.RpcError as e:
+        log(ERROR, "GRPC error occurred: %s", str(e))
+    finally:
+        channel.close()
+
+
+def run_clientappio_api_grpc(
+    address: str = "0.0.0.0:9094",
+) -> tuple[grpc.Server, grpc.Server]:
+    """Run ClientAppIo API (gRPC-rere)."""
+    clientappio_servicer: grpc.Server = ClientAppIoServicer()
+    clientappio_add_servicer_to_server_fn = add_ClientAppIoServicer_to_server
+    clientappio_grpc_server = generic_create_grpc_server(
+        servicer_and_add_fn=(
+            clientappio_servicer,
+            clientappio_add_servicer_to_server_fn,
+        ),
+        server_address=address,
+        max_message_length=GRPC_MAX_MESSAGE_LENGTH,
+    )
+    log(INFO, "Starting Flower ClientAppIo gRPC server on %s", address)
+    clientappio_grpc_server.start()
+    return clientappio_servicer, clientappio_grpc_server
```

Comment:
@danieljanes @chongshenng IMO, the server code should be moved into `/src/py/flwr/client/supernode`

Reply:
I agree that `process.py` isn't the ideal location. But putting it in `flwr/client/supernode/app.py` creates a circular import error. Maybe it's better to put it under `flwr/client/app.py` where it gets called? Wdyt @danieljanes @tanertopal?

Reply:
Moved `run_clientappio_api_grpc()` to `flwr/client/app.py` as this seems most logical.